pkg/cloudprovider/types.go

```diff
@@ -36,21 +36,21 @@ type DriftReason string
 type CloudProvider interface {
 	// Create launches a machine with the given resource requests and requirements and returns a hydrated
 	// machine back with resolved machine labels for the launched machine
-	Create(context.Context, *v1alpha5.Machine) (*v1alpha5.Machine, error)
+	Create(context.Context, *v1beta1.NodeClaim) (*v1beta1.NodeClaim, error)
 	// Delete removes a machine from the cloudprovider by its provider id
-	Delete(context.Context, *v1alpha5.Machine) error
+	Delete(context.Context, *v1beta1.NodeClaim) error
 	// Get retrieves a machine from the cloudprovider by its provider id
-	Get(context.Context, string) (*v1alpha5.Machine, error)
+	Get(context.Context, string) (*v1beta1.NodeClaim, error)
 	// List retrieves all machines from the cloudprovider
-	List(context.Context) ([]*v1alpha5.Machine, error)
+	List(context.Context) ([]*v1beta1.NodeClaim, error)
 	// GetInstanceTypes returns instance types supported by the cloudprovider.
 	// Availability of types or zone may vary by provisioner or over time.  Regardless of
 	// availability, the GetInstanceTypes method should always return all instance types,
 	// even those with no offerings available.
-	GetInstanceTypes(context.Context, *v1alpha5.Provisioner) ([]*InstanceType, error)
-	// IsMachineDrifted returns whether a machine has drifted from the provisioning requirements
+	GetInstanceTypes(context.Context, *v1beta1.NodePool) ([]*InstanceType, error)
+	// IsDrifted returns whether a machine has drifted from the provisioning requirements
 	// it is tied to.
-	IsMachineDrifted(context.Context, *v1alpha5.Machine) (DriftReason, error)
+	IsDrifted(context.Context, *v1beta1.NodeClaim) (DriftReason, error)
```

Comment:
Did we discuss making this IsDrifted() rather than IsNodeClaimDrifted?

Reply:
Discussed offline: Agreed that `IsDrifted` is a reasonable name for the interface method